syncd/scripts/syncd_init_common.sh

```diff
@@ -119,74 +119,175 @@ config_syncd_cisco_8000()
-      fi
-
-      #sync the file system
-      sync
-
-      # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
-      cp -f /tmp/sai.profile /var/run/sswsyncd/
-      cp -f /tmp/*.bcm /var/run/sswsyncd/
-
-      if [ -f "/tmp/sai.profile" ]; then
-          CMD_ARGS+=" -p /tmp/sai.profile"
-      elif [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+                else
+                   echo "        $line" >> $to_file
+                   merged_cnt+=1
+                fi
+            fi
+        fi
+    done < $from_file
 
-    else
+    if [ $merged_cnt -gt 0 ]; then
+         sed -i "/# Start of/a \    global:" $to_file
+         sed -i "/# Start of/a \  0:" $to_file
+         sed -i "/# Start of/a \bcm_device:" $to_file
+         sed -i "/# Start of/a \---" $to_file
+    fi
+    echo "# End of $message" >> $to_file
+    if [ $merged_cnt -gt 0 ]; then
+       sed -i "/# End of/i \..." $to_file
+    fi
+    echo "Merged $from_file to $to_file"
+}
 
-      if [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=$(find $HWSKU_DIR -name '*.bcm' -not -path "*/phy/*")
+    PLT_CONFIG_YML=$(find $HWSKU_DIR -name '*.yml')
+
+    if [ ! -z "$PLT_CONFIG_BCM" ] && [ -f $PLATFORM_DIR/common_config_support ] ; then
+       cp -f $HWSKU_DIR/*.config.bcm /tmp
+       cp -f /etc/sai.d/sai.profile /tmp
+       CONFIG_BCM=$(find /tmp -name '*.bcm')
+       SAI_PROFILE=$(find /tmp -name 'sai.profile')
+       sed -i 's+/usr/share/sonic/hwsku+/tmp+g' $SAI_PROFILE
+
+       #Get first three characters of chip id
+       readline=$(grep '0x14e4' /proc/linux-kernel-bde)
+       chip_id=${readline#*0x14e4:0x}
+       chip_id=${chip_id::3}
+       COMMON_CONFIG_BCM=$(find $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id} -maxdepth 1 -name '*.bcm')
+
+       if [ -f $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id}/*.bcm ]; then
+          for file in $CONFIG_BCM; do
+             merge_config_bcm_files $file $COMMON_CONFIG_BCM "chip common properties"
+          done
+          echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
+       fi
+       #sync the file system 
+       sync
+
+       # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
+       cp -f /tmp/sai.profile /var/run/sswsyncd/
+       cp -f /tmp/*.bcm /var/run/sswsyncd/
+    fi
+
+    if [ ! -z "$PLT_CONFIG_YML" ] && [ -f $PLATFORM_DIR/common_config_support ]; then
+       cp -f $HWSKU_DIR/*.config.yml /tmp
+       cp -f /etc/sai.d/sai.profile /tmp
+       CONFIG_YML=$(find /tmp -name '*.yml')
```

Comment:
@geans-pin what about this config.bcm file? It doesn't have .yml suffix, but it is yaml format, how to deal with it? https://github.com/sonic-net/sonic-buildimage/blob/b73d613bf581076192dd0150cb35d6d2de6645b1/device/arista/x86_64-arista_7060dx5_64s/Arista-7060DX5-64S/th4-a7060dx5-64s.config.bcm#L4

Reply:
We use the file name .yml suffix to identify the file format. Can you change the sai.profile and *.config.bcm to *.yml as following BRCM SDK rule ?